src/fsharp/NameResolution.fs

```diff
@@ -2475,7 +2476,15 @@ let rec ResolveExprLongIdentPrim sink (ncenv: NameResolver) first fullyQualified
                     | Exception e -> typeError := Some e; None
 
                 | true, res ->
-                    Some (FreshenUnqualifiedItem ncenv m res, [])
+                    let fresh = FreshenUnqualifiedItem ncenv m res, []
+                    match fresh |> fst with
```

Comment:
Explicit calls to `fst` are not normally used if there is a chance to use a tuple pattern instead.  S try

                    let (item, _) as fresh = FreshenUnqualifiedItem ncenv m res, []
                    match item with

Reply:
ok

Reply:
I will look at the failures today, I have no idea why they occur.